extensions/ql-vscode/src/remote-queries/export-results.ts

```diff
@@ -95,3 +97,32 @@ async function exportResultsToGist(
     }
   }
 }
+
+/**
+ * Converts the results of a remote query to markdown and saves the files locally
+ * in the query directory (i.e. the workspace directory where query results and logs are saved).
+ */
+async function exportResultsToLocalMarkdown(
+  queryHistoryManager: QueryHistoryManager,
+  queryHistoryItem: QueryHistoryInfo,
+  query: RemoteQuery,
+  analysesResults: AnalysisResults[]
+) {
+  const markdownFiles = generateMarkdown(query, analysesResults, 'local');
+  const basePath = await queryHistoryManager.getQueryHistoryItemDirectory(
+    queryHistoryItem
+  );
+  for (const markdownFile of markdownFiles) {
+    const filePath = path.join(basePath, `${markdownFile.fileName}.md`);
+    await fs.writeFile(filePath, markdownFile.content.join('\n'), 'utf8');
```

Comment:
Have you tested whether this works if you export multiple times? I think it should, but can't be bothered checking the docs 😬

Reply:
Oh, right...one thing we're going to need to think about is exporting multiple times with different selections. Presumably, we are going to want to delete the old results before creating new ones. Is this something we can implement later?

Reply:
Yeah I think we can worry about that a bit later on!

I specifically meant running the export multiple times e.g. maybe I forgot I already exported stuff and I tried it again.

Reply:
Exporting the same thing multiple times works fine!

Reply:
I guess we technically overwrite the markdown files every time though 🤔 Would that be a problem? Perhaps we should check if the files exist first?

Reply:
> Exporting the same thing multiple times works fine!

My concern is about exporting a slightly different selection each time. Since selection isn't implemented yet, we don't need to worry about it until later, but this is something we should keep in mind.